src/python/pybind11/src/pyxrt.cpp

```diff
@@ -33,263 +33,264 @@ namespace py = pybind11;
-                     })
-    .def("group_id", &xrt::kernel::group_id)
-    .def("write_register", &xrt::kernel::write_register)
-    .def("read_register", &xrt::kernel::read_register);
+    py::class_<xrt::run>(m, "run", "Represents one execution of a kernel")
+        .def(py::init<>())
+        .def(py::init<const xrt::kernel &>())
+        .def("start", [](xrt::run& r){
+                          r.start();
+                      }, "Start one execution of a run")
+        .def("set_arg", [](xrt::run& r, int i, xrt::bo& item){
+                            r.set_arg(i, item);
+                        }, "Set a specific kernel global argument for a run")
+        .def("set_arg", [](xrt::run& r, int i, int& item){
+                            r.set_arg<int&>(i, item);
+                        }, "Set a specific kernel scalar argument for this run")
+        .def("wait", ([](xrt::run& r, unsigned int timeout_ms)  {
+                          return r.wait(timeout_ms);
+                      }), "Wait for the specified milliseconds for the run to complete")
+        .def("state", &xrt::run::state, "Check the current state of a run object")
+        .def("add_callback", &xrt::run::add_callback, "Add a callback function for run state");
+
+    py::class_<xrt::kernel> pyker(m, "kernel", "Represents a set of instances matching a specified name");
+
+    py::enum_<xrt::kernel::cu_access_mode>(pyker, "cu_access_mode", "Compute unit access mode")
+        .value("exclusive", xrt::kernel::cu_access_mode::exclusive)
+        .value("shared", xrt::kernel::cu_access_mode::shared)
+        .value("none", xrt::kernel::cu_access_mode::none)
+        .export_values();
+
+
+    pyker.def(py::init([](const xrt::device& d, const xrt::uuid& u, const std::string& n,
+                          xrt::kernel::cu_access_mode m) {
+                           return new xrt::kernel(d, u, n, m);
+                       }))
+        .def("__call__", [](xrt::kernel& k, py::args args) -> xrt::run {
+                             int i = 0;
+                             xrt::run r(k);
+
+                             for (auto item : args) {
+                                 try {
+                                     r.set_arg(i, item.cast<xrt::bo>());
+                                 }
+                                 catch (std::exception e) {  }
+
+                                 try {
+                                     r.set_arg<int>(i, item.cast<int>());
+                                 }
+                                 catch (std::exception e) {  }
+
+                                 i++;
+                             }
+
+                             r.start();
+                             return r;
+                         })
+        .def("group_id", &xrt::kernel::group_id, "Get the memory bank group id of an kernel argument")
+        .def("write_register", &xrt::kernel::write_register)
+        .def("read_register", &xrt::kernel::read_register);
```

Comment:
We are considering deprecating read/write_register from managed kernel APIs (xrt::kernel).   It would be better if we didn't wrap these APIs in python.  read/write_register is supported through custom IPs (xrt::ip).

Reply:
Makes sense, dropped it from the pybind11 doc.